File: Java/wellmind-app/src/main/java/com/wellmind/dto/usuario/CreateUsuarioDTO.java
package com.wellmind.dto.usuario;

import jakarta.validation.constraints.*;
import lombok.*;

import java.time.LocalDate;

/**
 * DTO para criar um novo usuário
 */
@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class CreateUsuarioDTO {

    @NotBlank(message = "{validation.nome.not.blank}")
    @Size(min = 3, max = 100, message = "{validation.nome.size}")
    private String nome;

    @Email(message = "{validation.email.invalid}")
    @NotBlank(message = "{validation.email.not.blank}")
    private String email;

    @NotBlank(message = "{validation.senha.not.blank}")
    @Size(min = 8, max = 255, message = "{validation.senha.size}")
    @Pattern(regexp = "^(?=.*[A-Z])(?=.*[0-9])(?=.*[!@#$%^&*]).*$",
            message = "{validation.senha.pattern}")
    private String senha;

    @NotNull(message = "{validation.datanascimento.not.null}")
    @PastOrPresent(message = "{validation.datanascimento.past}")
    private LocalDate dataNascimento;

    @NotBlank(message = "{validation.genero.not.blank}")
    private String genero; // Masculino, Feminino, Outro, Prefiro não informar

    @NotBlank(message = "{validation.telefone.not.blank}")
    @Size(min = 11, max = 20, message = "{validation.telefone.size}")
    private String telefone;
}

File: DotNet/WellMindApi/src/WellMindApi.Application/DTOs/Usuarios/CreateUsuarioDto.cs
namespace WellMindApi.Application.DTOs.Usuarios;

/// <summary>
/// DTO para criação de usuário
/// </summary>
public record CreateUsuarioDto
{
    public string Nome { get; init; } = string.Empty;
    public string Email { get; init; } = string.Empty;
    public string Senha { get; init; } = string.Empty;
    public DateTime DataNascimento { get; init; }
    public string? Genero { get; init; }
    public string? Telefone { get; init; }
}
